### ISSUE TITLE
unrecognized global Promise when setting env node and browser

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#10837]
**Tell us about your environment**

* **ESLint Version:** v5.5.0
* **Node Version:** v10.9.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```js
  "eslintConfig": {
    "extends": "eslint:recommended",
    "parserOptions": {
      "ecmaVersion": 8
    },
    "env":{
      "browser": true,
      "node": true
    }
  },
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function timeout(delay) {
  return new Promise((resolve, reject) => {
      setTimeout(() => {
          try {
              resolve(1)
          } catch (e) {
              reject(0)
          }
      }, delay)
  })
}
```

```bash

```

**What did you expect to happen?**
no error report, since `ecmaVersion` is 8.

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
53:14  error  'Promise' is not defined          no-undef
```

**Comment by Aladdin-ADD** [comment]
oh, I missed it, thanks! !
